workspaces/templates-lib/packages/infra-aws/src/awsUserUtils.ts

```diff
@@ -0,0 +1,193 @@
+import assert from 'assert';
+import {
+  AWSConfiguration,
+  AWSRegion,
+  AWSProfileConfig,
+  AWSAPIKeyUserConfig,
+  AWSEnvironmentVariableUserConfig,
+} from './types/awsAccount';
+import AWS from 'aws-sdk';
+
+export async function getAWSUserFromEnvironmentVariables(): Promise<AWS.Credentials> {
+  assert(process.env.AWS_ACCESS_KEY_ID, 'AWS_ACCESS_KEY_ID not defined.');
+  assert(
+    process.env.AWS_SECRET_ACCESS_KEY,
+    'AWS_SECRET_ACCESS_KEY not defined'
+  );
+  const region = process.env.AWS_REGION || process.env.AWS_DEFAULT_REGION;
+  assert(region, 'Neither AWS_REGION nor AWS_DEFAULT_REGION are defined.');
+  const credentials = new AWS.EnvironmentCredentials('AWS');
+  await credentials.getPromise();
+  AWS.config.credentials = credentials;
+
+  AWS.config.update({ region });
+
+  return credentials;
+}
+/**
+ * Obtains AWS user credentials from container environment variables for ECS containers.
+ */
+export async function getAWSUserFromContainerEnvironment(): Promise<AWS.ECSCredentials> {
+  const ecsCredentials = new AWS.ECSCredentials({
+    httpOptions: { timeout: 5000 },
+    maxRetries: 10, // retry 10 times
+  });
+  await ecsCredentials.getPromise();
+
+  AWS.config.credentials = ecsCredentials;
+
+  if (!process.env.AWS_REGION) {
+    throw new Error(
+      'AWS region environment variable ("AWS_REGION") not defined for ECS task.'
+    );
+  }
+
+  AWS.config.update({ region: process.env.AWS_REGION });
+
+  return ecsCredentials;
+}
+
+export async function getAWSUserFromDefaultLocalProfile(): Promise<AWS.Credentials> {
+  let credentials = new AWS.SharedIniFileCredentials();
+
+  // if no access key is found, try loading process_credentials
+  if (!credentials.accessKeyId) {
+    credentials = new AWS.ProcessCredentials();
+    await credentials.refreshPromise();
+  }
+
+  AWS.config.credentials = credentials;
+  // see https://github.com/aws/aws-sdk-js/pull/1391
+  process.env.AWS_SDK_LOAD_CONFIG = 'true';
```

Comment:
Interesting that this is here. After L55-L56, I'm not sure this would have an effect.

Reply:
Yeah this definitely doesn't make sense here. I've moved this now before instantiating the credentials - but I wonder if that's required? https://github.com/goldstack/goldstack/pull/102/files#diff-e43f64484d679c0de71cca9f787c4dc4d20e104fbd0faada8b38efc275670918R53